DatadogRUM/Tests/RUMTests.swift

```diff
@@ -74,6 +74,23 @@ class RUMTests: XCTestCase {
         XCTAssertTrue(RUM._internal.isEnabled(in: core))
     }
 
+    func testWhenEnabled_currentSessionStartsAsNil() {
+        // Given
+        let expectation = XCTestExpectation(description: "currentSessionID called")
+
+        // When
+        RUM.enable(with: config, in: core)
+        XCTAssertNotNil(core.get(feature: RUMFeature.self))
+
+        // Then
+        RUMMonitor.shared().getCurrentSessionID {
+            XCTAssertNil($0)
+            expectation.fulfill()
+        }
+
+        wait(for: [expectation], timeout: 0.1)
+    }
+
```

Comment:
blocking/ This test is false positive. There is no such behaviour as "current session starts as `nil`" after https://github.com/DataDog/dd-sdk-ios/pull/1594

One problem is that `RUMMonitor.shared()` is missing the `core`:
```swift
RUMMonitor.shared(in: core)
```

However, if we pass the `core` reference this test will fail because we use `FeatureRegistrationCoreMock` in this file and that only offers feature registration capability, see: https://github.com/DataDog/dd-sdk-ios/blob/a43c63dd422ffd050ed24a1ac1dfd7b156f063c2/TestUtilities/Mocks/CoreMocks/FeatureRegistrationCoreMock.swift#L9-L23

If we change it to `PassthroughCoreMock` as suggested, it will keep failing because that [mock doesn't support `get<T>(feature:)`](https://github.com/DataDog/dd-sdk-ios/blob/a43c63dd422ffd050ed24a1ac1dfd7b156f063c2/TestUtilities/Mocks/CoreMocks/PassthroughCoreMock.swift#L94) required in [`RUMMonitor.shared(in:)](https://github.com/DataDog/dd-sdk-ios/blob/a43c63dd422ffd050ed24a1ac1dfd7b156f063c2/DatadogRUM/Sources/RUMMonitor.swift#L35).

After all, I think the problem is because we try to test `RUMMonitor` in `RUMTests.swift`, which is only configured to test `RUM.*` configuration and no behaviours. My recommendation is to remove this test from here and cover this flow in `RUMMonitorTests.swift`, which depends on real instance of the core.

Reply:
I think I'll just delete this test then, and add a test to RUMMonitorTests that checks that the session is `nil` after a call to `stopSession`.

Thoughts?